### PR TITLE
Implement sitemap with basic dynamic routing

### PR DESCRIPTION
**Pull request by renzol2**
## What this pr does
Implements the pages sitemap as planned in the planning doc
Creates constants in `/pages/api/constants` for frontend dev purposes
Displays a simple routing example from the home page

## Usage
Click the links to navigate (there's no way to go back to the home page yet, that will be fixed later with a navbar)

## Testing
¯\\_(ツ)_/¯

### CODE REVIEW

**Review by AlpriElse — CHANGES_REQUESTED** [review]
Instead of using the constants directly in the components, could you create a new file `api/client.js`?

In that file you would have functions like `fetchUsers()`, `fetchBuilds()`, etc that fill return the constants.

Also, those function would have to return a Promise:
```javascript
function fetchUsers() {
  return new Promise((resolve, reject) => {
    resolve(the data you want to return)
  })
}
```

The reason for doing this is so that when we connect the API, we only have to worry about changing those functions to return values from the API instead of constants rather than changing the components themselves.

Eventually, those functions would look something like this:
```javascript
function fetchUsers(query) {
  return axios.get(HTTP_ENDPOINT, { query })
}
```

Also, you would also have to set them up to handle a promise rather than a pure function. So, each component would have something like this (unless we want to set up some global context).
```javascript
const [users, setUsers] = useState(null)
useEffect(() => {
  fetchUsers.then(res => setUsers(res))
}, [])
```

File: pages/index.js [hunk 1]
```diff
@@ -1,65 +1,67 @@
...
-            className={styles.card}
-          >
-            <h3>Examples &rarr;</h3>
-            <p>Discover and deploy boilerplate example Next.js projects.</p>
-          </a>
+      <h2>Users</h2>
+      <ul>
+        {USERS.map(user => (
+          <li>
+            <Link href="/[user]" as={`/${user.name}`}>
```
Comment (AlpriElse) [inline pages/index.js:44]: Just wondering, why are users under `/:user` and tags under `/tags/:tag`

Comment (renzol2) [reply pages/index.js:44]: Product Hunt, Instagram, and Twitter all have user profiles under `/:user`, so I figured for familiarity reasons I would follow the same format. By the same lines, Instagram/Twitter and Product Hunt use hashtags/topics under `/hashtag/:hashtag` and `/topic/:topic` respectively. Let me know if you think that's weird though :sweat_smile:

**Comment by AlpriElse** [comment]
Also, you don't have to do it for this one, but it usually helps if you put screenshots within the pr description for UI type changes just for context. Cause I'm lazy and don't want to look at the preview deploy

**Comment by renzol2** [comment]
I've added the `api/client.js` file with the functions as you mentioned - let me know if it looks good to you. Also, Next.js offers a [solution](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps) for pre-populating the page with data from the server, so I've used that instead of `useEffect`. I'll have to research more into it though!

And yes, I will add screenshots when more meaningful UI is done!

**Pull request merged**